rxjava-support/src/main/java/tech/picnic/rx/RetryStrategy.java

```diff
@@ -129,11 +132,26 @@ public static Builder onlyIf(Predicate<Throwable> predicate) {
      * @return this {@link RetryStrategy.Builder}; useful for method chaining
      */
     public Builder exponentialBackoff(Duration initialDelay) {
+      return customBackoff(getExponentialDelays(initialDelay));
+    }
+
+    /**
+     * Configures the use of an exponentially increasing backoff delay up to a maximum.
+     *
+     * @param initialDelay the delay before the first retry
+     * @param maxBackoffDelay the maximum delay between retries
+     * @return this {@link RetryStrategy.Builder}; useful for method chaining
+     */
+    public Builder boundedExponentialBackoff(Duration initialDelay, Duration maxBackoffDelay) {
       return customBackoff(
-          Flowable.just(2)
-              .repeat()
-              .scan(initialDelay.toMillis(), (i, j) -> i * j)
-              .map(Duration::ofMillis));
+          getExponentialDelays(initialDelay).map(delay -> min(asList(delay, maxBackoffDelay))));
+    }
+
+    private static Flowable<Duration> getExponentialDelays(Duration initialDelay) {
+      return Flowable.just(2)
+          .repeat()
+          .scan(initialDelay.toMillis(), LongMath::saturatedMultiply)
```

Comment:
`saturatedMultiply` is `@Beta` annotated, but given its been around since version Guava `20.0` I guess it's safe?

Reply:
We use tons of `@Beta` things in guava :) (e.g. `ImmutableMap#toImmutableMap`), so I wouldn't worry to much about that

Reply:
Hmm, but our build [should](https://github.com/PicnicSupermarket/reactive-support/blob/336758dcdb6942aeaf92c8f0c66316526edb3fdf/pom.xml#L782) have failed because of this :thinking:. Will make a note to investigate later.